### PR TITLE
Improve completion item sorting in `order by` context

### LINKED ISSUE #33882: Improve completion sorting in `order by` context
**Description:**
The expression with basic types should have the highest precedence. 
Currently there is no such order.

**Steps to reproduce:**
```ballerina
function findTop10MedalWinners() returns error? {
    int rank = ;
    string[]|error countries = from var stats in check getMedalStats() 
    order by <cursor>
    
}
```

### PR DESCRIPTION
**Pull request by TharushiJay**
## Purpose
This PR improves completion item sorting in order-by context giving priority to basic types.

Fixes #33882

## Samples
Screenshot 2021-12-13 at 16 57 52

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java [hunk 1]
```diff
@@ -1801,6 +1802,29 @@ public static Boolean withinLockStatementNode(BallerinaCompletionContext context
...
+    /**
+     * Check whether the completion item type belongs to the types list passed.
+     *
+     * @param lsCItem   Completion item
+     * @param typesList List of types           
+     * @return {@link Boolean}
+     */
+    public static boolean isTypeCompletionItem(LSCompletionItem lsCItem, List<TypeDescKind> typesList) {
+
+        if (lsCItem.getType() != LSCompletionItem.CompletionItemType.SYMBOL) {
```
Comment (malinthar) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java:1814]: We are ignoring `LSCompletionItem.CompletionItemType.TYPE` here. Is it intended? Or should we change the name of the method to something like `isOfType`

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java [hunk 2]
```diff
@@ -1801,6 +1802,29 @@ public static Boolean withinLockStatementNode(BallerinaCompletionContext context
...
     }
 
+    /**
+     * Check whether the completion item type belongs to the types list passed.
+     *
+     * @param lsCItem   Completion item
+     * @param typesList List of types           
+     * @return {@link Boolean}
+     */
+    public static boolean isTypeCompletionItem(LSCompletionItem lsCItem, List<TypeDescKind> typesList) {
```
Comment (IMS94) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java:1812]: IMO this method name is a bit misleading. It sounds like checking if the completion item is a "type completion item". Shall we rename it to something like `isCompletionItemOfType()`. WDYT?

**Review by malinthar — COMMENTED** [review]
Changes look good. I have a small concern about the method name. It can be misleading.

**Review by IMS94 — CHANGES_REQUESTED** [review]

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java (hunk 1 above)
Comment (TharushiJay) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java:1816]: +1. Changed the method name to `isCompletionItemOfType()`.

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java (hunk 2 above)
Comment (TharushiJay) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java]: +1. Changed the method name to `isCompletionItemOfType()`.

**Review by malinthar — APPROVED** [review]
LGTM

**Review by IMS94 — APPROVED** [review]
LGTM!

**Review by gayalkuruppu — APPROVED** [review]
LGTM

**Pull request merged**